services/frontend-service/pkg/interceptors/interceptors.go

```diff
@@ -123,6 +124,21 @@ func GoogleIAPInterceptor(
 	httpHandler(w, req)
 }
 
+func checkPolicy(httpCtx context.Context, w http.ResponseWriter, req *http.Request, userGroup string) context.Context {
+	policy, err := auth.ReadRbacPolicy(true, "/etc/policy.csv")
+	if err != nil {
+		http.Error(w, "unable to access RBAC policy to validate login", http.StatusBadRequest)
+		return nil
+	}
+	for _, policyGroup := range policy.Groups {
+		if policyGroup.Group == userGroup {
+			auth.WriteUserRoleToHttpHeader(req, policyGroup.Role)
```

Comment:
I am not sure because currently we only expect one so for now I keep over writing but I am unsure if I should change the role from a string into a list

Reply:
The JWT-Claim of the user could contain multiple groups  so it could also contain multiple roles which are assigned to the different groups is what I was thinking of.

But maybe this is just an edge case which will not happen 👍

Reply:
I feel that that is a bigger issue that we should maybe decide, since some roles could have differing policies, but that can be expanded to be allowed later?